### ISSUE TITLE
MSRV on `sha2` v0.10.3 (and v0.10.4) is broken

### ISSUE THREAD

**Issue by striezel** [RustCrypto/hashes#394]
When trying to build a project that depends on `sha2` v0.10.3 with rustc 1.48, one gets error messages from cargo. This appears to be an instance of <https://github.com/rust-lang/cargo/issues/10954>, which means the `sha2` crate simply needs to be re-uploaded with stable cargo or a newer nightly cargo.

The error message is:
```
   Downloaded sha2 v0.10.3
error: failed to parse manifest at `/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/sha2-0.10.3/Cargo.toml`

Caused by:
  feature `resolver` is required

  this Cargo does not support nightly features, but if you
  switch to nightly channel you can add
  `cargo-features = ["resolver"]` to enable this feature
```

**Comment by striezel** [comment]
@newpavlov: Thanks for the attempted fix in #399. :+1:

Unfortunately, this issue is not fixed in sha2 v0.10.4, the new release still contains a `resolver = "2"` line in the generated Cargo.toml: https://docs.rs/crate/sha2/0.10.4/source/Cargo.toml
Error message is basically still the same as shown above, except with a newer version number.

**Comment by striezel** [comment]
The issues seems to be fixed by #401, i. e. in `sha2` v0.10.5.
